File: codeforces/5.domino_piling/domino_piling.cpp
#include <iostream>

/* Author: José Rodolfo (jric2002) */

using namespace std;

int main() {
  unsigned short int m, n;
  cin >> m >> n;
  if (m >= 1 && m <= n && n <= 16) {
    unsigned short int max_num_dominoes;
    max_num_dominoes = (m * n) / 2;
    cout << max_num_dominoes << endl;
  }
  return 0;
}

File: codeforces/5.domino_piling/domino_piling.py
# Author: José Rodolfo (jric2002)
import math
input_data = input().split(" ")
input_data = [int(e) for e in input_data]
m, n = input_data
if (m >= 1 and m <= n and n <= 16):
  num_max_dominoes = math.floor((m * n) / 2)
  print(num_max_dominoes)